holoviews/plotting/bokeh/widgets.py

```diff
@@ -95,52 +93,80 @@ def create_widget(self, dim, holomap=None, editable=False):
         label, mapping = None, None
         if holomap is None:
             if dim.values:
+                if dim.default is None:
+                    default = dim.values[0]
+                elif dim.default not in dim.values:
+                    raise ValueError("%s dimension default %r is not in dimension values: %s"
+                                     % (dim, dim.default, dim.values))
+                else:
+                    default = dim.default
+                value = dim.values.index(default)
+
                 if all(isnumeric(v) for v in dim.values):
-                    values = dim.values
-                    labels = [unicode(dim.pprint_value(v)) for v in dim.values]
+                    values = sorted(dim.values)
+                    labels = [unicode(dim.pprint_value(v)) for v in values]
                     if editable:
-                        label = AutocompleteInput(value=labels[0], completions=labels,
+                        label = AutocompleteInput(value=labels[value], completions=labels,
                                                   title=dim.pprint_label)
                     else:
-                        label = Div(text='<b>%s</b>' % dim.pprint_value_string(labels[0]))
-                    widget = Slider(value=0, start=0, end=len(dim.values)-1, title=None, step=1)
-                    mapping = list(zip(values, labels))
+                        label = Div(text='<b>%s</b>' % dim.pprint_value_string(labels[value]))
+                    widget = Slider(value=value, start=0, end=len(dim.values)-1, title=None, step=1)
+                    mapping = list(enumerate(zip(values, labels)))
                 else:
                     values = [(v, dim.pprint_value(v)) for v in dim.values]
-                    widget = Select(title=dim.pprint_label, value=values[0][0],
+                    widget = Select(title=dim.pprint_label, value=values[value][0],
                                     options=values)
             else:
                 start = dim.soft_range[0] if dim.soft_range[0] else dim.range[0]
                 end = dim.soft_range[1] if dim.soft_range[1] else dim.range[1]
                 dim_range = end - start
                 int_type = isinstance(dim.type, type) and issubclass(dim.type, int)
-                if isinstance(dim_range, int) or int_type:
-                    step = 1
-                elif dim.step is not None:
+                if dim.step is not None:
                     step = dim.step
+                elif isinstance(dim_range, int) or int_type:
+                    step = 1
                 else:
                     step = 10**((round(math.log10(dim_range))-3))
+
+                if dim.default is None:
+                    default = start
+                elif (dim.default < start or dim.default > end):
+                    raise ValueError("%s dimension default %r is not in the provided range: %s"
+                                     % (dim, dim.default, (start, end)))
```

Comment:
Same question about it not already being validated? Isn't ``start`` and ``end`` defined by the dimension range?

Reply:
True it is now